compiler/qsc_passes/src/measurement.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_hir::{
+    hir::{Attr, CallableDecl, Item, ItemKind, Package, SpecBody, SpecGen},
+    ty::{Prim, Ty},
+    visit::Visitor,
+};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum Error {
+    #[error("a measurement should take at least one Qubit as argument")]
```

Comment:
nitpick: for this and other errors in the file, since `Qubit` and `Result` are types, I would phrase it in a way that leans into that, with patterns like "a custom measurement should take at least one argument of type `Qubit`" or similar. Also, worth using "custom measurement" or perhaps even "a callable with the measurement attribute" to make it clear that we are talking specifically about use of the `@Measurement` feature and not the concept of measurement in general.